README.md

```diff
@@ -889,19 +871,19 @@ Resolves with the id token string retrieved from storage if it exists. Devs shou
 
 Resolves with the access token string retrieved from storage if it exists. Devs should prefer to consult the synchronous results emitted from subscribing to the [authStateManager.subscribe](#authstatemanagersubscribehandler).
 
-### `handleAuthentication()`
+### `parseAndStoreTokensFromUrl()`
 
 > :hourglass: async
 
 Parses tokens from the redirect url and stores them.
 
-### `setFromUri(uri?)`
+### `setFromUri(fromUri?)`
 
-Store the current URL state before a redirect occurs. If a relative path is passed it will be converted to an absolute URI before storage.
+Stores the current URL state before a redirect occurs. By default it stores `window.location.href`.
 
-### `getFromUri(relative?)`
+### `getFromUri()`
 
-Returns the stored URI string stored by [setFromUri](#setfromuriuri) and removes it from storage.  A relative uri is returned if `relative` is true.
+Returns the stored URI string stored by [setFromUri](#setfromuriuri) and removes it from storage. By default it returns `window.location.origin`.
```

Comment:
As long as we are updating this API, do you think it makes sense to separate "get" from "remove"? It's not difficult to call `getFromUri` and then `removeFromUri` but it is confusing/subtle that `getFromUri` is a one-time use. Generally I would expect a getter function to not produce side-effects.

It might also be nice if these functions were named or namespaced in a way that it was clear they are related to the **redirect flow**. There is a family of functions related to this flow: signInWithRedirect, parse/store tokens, get/set/remove uri, and isLoginRedirect.

Reply:
Per the name scope of the *redirect flow methods*, I think we can scope it into a separate task, as `isLoginRedirect ` right now belongs to `token` scope, make changes like this would introduce breaking changes anyway. 

Also, it might make sense to introduce a layer of shim to define the public APIs for the downstream SDKs, the benefit is that we don't always need to make major version update on downstream sides if we want to change scopes in the upstream. 

Per the `getFromUri `, I think we almost always want users to clear the stored value when get it, otherwise a one-liner call on sessionStorage should achieve the same effect. How about just rename the method to make the purpose clearer?

Reply:
Why not add the new methods to the `token` namespace then?

For `getFromUri` I should be able to read the value and make decisions without actually breaking the flow. The one-time usage is too subtle. Yes, they probably should clear it, but it won't really cause a problem if they don't, since the flow always begins with a `setFromUri`. I think these methods should be simple CRUD, the only part we are abstracting is **where** this value is stored.